test/pipelines/pipelines.test.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable func-names */
+import { expect } from 'chai';
+import { closeApplication, createApplication } from '../application';
+import { DEFAULT_GLOBAL_TIMEOUT } from '../constants';
+import {
+  PRIMARY_PIPELINE,
+  SECONDARY_PIPELINE,
+  SECOND_ALGORITHM,
+  ALGORITHMS_PIPELINES,
+} from '../fixtures/pipelines/pipelines';
+import { mochaAsync } from '../utils';
+import {
+  addPipeline,
+  clickAddButton,
+  addAlgorithmToPipeline,
+  clickSavePipelineButton,
+  checkPipelineRowLayout,
+  clickEditPipelineButton,
+  removeAlgorithmPipeline,
+  clickDeletePipelineButton,
+  getNumberOfPipelines,
+  editNamePipeline,
+  editDescriptionPipeline,
+} from './utils';
+
+describe('Pipelines Scenarios', function () {
+  this.timeout(DEFAULT_GLOBAL_TIMEOUT);
+  let app = null;
+
+  beforeEach(
+    mochaAsync(async () => {
+      app = await createApplication({
+        database: {
+          algorithms: ALGORITHMS_PIPELINES,
+        },
+        responses: { showMessageDialogResponse: 1 },
+      });
+      const { client } = app;
+      await client.goToPipelines();
+    }),
+  );
+
+  afterEach(() => {
+    return closeApplication(app);
+  });
+
+  it(
+    'Adds, Edits, Deletes a pipeline',
+    mochaAsync(async () => {
+      const { client } = app;
+
+      const nbrPipelinesPrev = await getNumberOfPipelines(client);
+
+      // add pipeline
+      await clickAddButton(client);
+      await addPipeline(client, PRIMARY_PIPELINE);
+
+      for (const algo of ALGORITHMS_PIPELINES) {
+        await addAlgorithmToPipeline(client, algo);
+      }
+
+      await clickSavePipelineButton(client);
+      await checkPipelineRowLayout(client, PRIMARY_PIPELINE);
+
+      const nbrPipelinesAfterAdd = await getNumberOfPipelines(client);
+      expect(nbrPipelinesAfterAdd - nbrPipelinesPrev).to.be.equal(1);
+
+      // edit pipeline
+      await clickEditPipelineButton(client, PRIMARY_PIPELINE);
+      await removeAlgorithmPipeline(
+        client,
+        ALGORITHMS_PIPELINES.findIndex((x) => x.id === SECOND_ALGORITHM.id),
+      );
+      await editNamePipeline(client, SECONDARY_PIPELINE);
+      await editDescriptionPipeline(client, SECONDARY_PIPELINE);
+      await clickSavePipelineButton(client);
+
```

Comment:
I thing something important to check here is whether the changes are persisted. Eg if you quit the  screen and comeback are the changes actually saved.

Reply:
And check if the added algorithms are in the correct order.